src/hotspot/share/opto/inlinetypenode.hpp

```diff
@@ -100,15 +101,21 @@ class InlineTypeNode : public TypeNode {
   Node* get_is_init() const { return in(IsInit); }
   void  set_is_init(PhaseGVN& gvn) { set_req(IsInit, gvn.intcon(1)); }
   void  set_is_buffered() { _is_buffered = true; }
+  bool  is_buffered() { return _is_buffered; }
 
   // Inline type fields
-  uint          field_count() const { return req() - Values; }
-  Node*         field_value(uint index) const;
+  virtual uint  field_count() const { return req() - Values; }
+  virtual Node* field_value(uint index) const;
+  uint          field_index(int offset) const;
+
   Node*         field_value_by_offset(int offset, bool recursive = false) const;
-  void      set_field_value(uint index, Node* value);
-  void      set_field_value_by_offset(int offset, Node* value);
+  void          set_field_value(uint index, Node* value);
+  void          set_field_value_by_offset(int offset, Node* value);
   int           field_offset(uint index) const;
-  uint          field_index(int offset) const;
+  bool          is_multifield(uint index) const;
+  bool          is_multifield_base(uint index) const;
+  int           secondary_field_count(uint index) const;
```

Comment:
Rename to `secondary_fields_count` ? Or is it possible getting the count from the field type like `type->bundle_size()`, so that this method can be removed?

Reply:
BTW, methods like `is_multifield, is_multifield_base` can be removed. `secondary_fields_count() > 1` can be used as a replacement to `is_multifield_base()`, right?

Reply:
> BTW, methods like `is_multifield, is_multifield_base` can be removed. `secondary_fields_count() > 1` can be used as a replacement to `is_multifield_base()`, right?

It should add value to still keep is_multifield and is_multifield_base since it allows skipping over synthetic fields (non-base) at places.

Reply:
I'm sorry that I don't understand quite well. At least, `is_multifield()` is not used now. So I guess it can be removed?